react/features/chrome-extension-banner/components/ChromeExtensionBanner.js

```diff
@@ -0,0 +1,255 @@
+
+/**
+ * The type of the React {@code PureComponent} props of {@link ChromeExtensionBanner}.
+ */
+type Props = {
+
+    /**
+     * The url of the chrome extension
+     */
+    chromeExtensionUrl: string,
+
+    /**
+     * An array containing info for identifying a chrome extension
+     */
+    chromeExtensionsInfo: Array<Object>,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function,
+};
+
+/**
+ * The type of the React {@link PureComponent} state of {@link ChromeExtensionBanner}.
+ */
+type State = {
+
+    /**
+     * Keeps the current value of dont show again checkbox
+     */
+    dontShowAgainChecked: boolean,
+
+    /**
+     * Tells whether user pressed install extension or close button.
+     */
+    closePressed: boolean,
+
+    /**
+     * Tells whether should show the banner or not based on extension being installed or not.
+     */
+    shouldShow: boolean,
+};
+
+/**
+ * Implements a React {@link PureComponent} which displays a banner having a link to the chrome extension.
+ * @class ChromeExtensionBanner
+ * @extends PureComponent
+ */
+class ChromeExtensionBanner extends PureComponent<Props, State> {
+    /**
+     * Initializes a new {@code ChromeExtensionBanner} instance.
+     *
+     * @param {Object} props - The read-only React {@code PureComponent} props with
+     * which the new instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            dontShowAgainChecked: false,
+            closePressed: false,
+            shouldShow: false
+        };
+
+        this._onClosePressed = this._onClosePressed.bind(this);
+        this._onInstallExtensionClick = this._onInstallExtensionClick.bind(this);
+        this._checkExtensionsInstalled = this._checkExtensionsInstalled.bind(this);
+        this._shouldNotRender = this._shouldNotRender.bind(this);
+        this._onDontShowAgainChange = this._onDontShowAgainChange.bind(this);
+    }
+
+    /**
+     * Executed on component update.
+     * Checks whether any chrome extension from the config is installed.
+     *
+     * @inheritdoc
+     */
+    async componentDidUpdate() {
+        const hasExtensions = await this._checkExtensionsInstalled();
+
+        if (
+            hasExtensions
+            && hasExtensions.length
+            && hasExtensions.every(ext => !ext)
+            && !this.state.shouldShow
+        ) {
+            this.setState({ shouldShow: true }); // eslint-disable-line
+        }
+    }
+
+    _onClosePressed: () => void;
+
+    /**
+     * Closes the banner for the current session.
+     *
+     * @returns {void}
+     */
+    _onClosePressed() {
+        this.setState({ closePressed: true });
+    }
+
+    _onInstallExtensionClick: () => void;
+
+    /**
+     * Opens the chrome extension page.
+     *
+     * @returns {void}
+     */
+    _onInstallExtensionClick() {
+        window.open(this.props.chromeExtensionUrl);
```

Comment:
If state['features/base/config'].chromeExtensionBanner is not configured we will try to open a window with undefined link.

Reply:
That's true, so probably I should set SHOW_CHROME_EXTENSION_BANNER defaulted to false in interface_config, so that we can enable it only for the environments where we will set correct values for `chromeExtensionBanner`/

Setting SHOW_CHROME_EXTENSION_BANNER  to false will prevent rendering the component alltogheter.

Reply:
You should make sure if SHOW_CHROME_EXTENSION_BANNER from interface config is enabled but the config.js setting is not set to act as SHOW_CHROME_EXTENSION_BANNER is disabled and probably print a warning saying that further configuration is needed, so users configuring this to be aware of the problems.

Reply:
fixed